tools/cbmc/proofs/HTTP/global_state_HTTP.c

```diff
@@ -325,10 +337,17 @@ IotHttpsRequestHandle_t allocate_IotRequestHandle() {
   IotHttpsRequestHandle_t pRequestHandle =
     safeMalloc(sizeof(struct _httpsRequest));
   if (pRequestHandle) {
-    uint32_t headerLen;
     pRequestHandle->pHttpsResponse = allocate_IotResponseHandle();
     pRequestHandle->pHttpsConnection = allocate_IotConnectionHandle();
+
+    size_t headerLen;
+    __CPROVER_assume(headerLen < CBMC_MAX_OBJECT_SIZE);
     pRequestHandle->pHeaders = safeMalloc(headerLen);
+    pRequestHandle->pHeadersEnd = pRequestHandle->pHeadersCur = pRequestHandle->pHeaders;
+    if (pRequestHandle->pHeaders) {
+      pRequestHandle->pHeadersEnd = pRequestHandle->pHeaders + headerLen - 1;
+    }
```

Comment:
This seems wrong again for similar reasons.

Reply:
As per our discussion, I have now added offsets that are non-deterministically assigned between 0 and `headerLen`, and are then added to `pHeaders`.